packages/shopify-api/docs/guides/oauth.md

```diff
@@ -1,22 +1,96 @@
 # Performing OAuth
 
+##### Table of content
+- [Supported types of OAuth Flow](#supported-types-of-oauth-flow)
+- [Token Exchange](#token-exchange)
+  - [Remix App](#remix-app)
+  - [Non-Remix App](#non-remix-app)
+  - [Detecting scope changes](#detecting-scope-changes)
+- [Authorization Code Grant Flow](#authorization-code-grant-flow)
+  - [Detecting scope changes](#detecting-scope-changes-1)
+- [After OAuth](#after-oauth)
+
+----------------------------------------------------------------------------------
 After you set up the library for your project, you'll be able to use it to interact with the APIs, and add your own functionality.
 The first thing your app will need to do is to get a token to access the Admin API by performing the OAuth process. Learn more about [OAuth on the Shopify platform](https://shopify.dev/docs/apps/auth/oauth).
 
-To perform OAuth, you will need to create two endpoints in your app:
+## Supported types of OAuth
+> [!TIP]
+> If you are building an embedded app, we **strongly** recommend using [Shopify managed installation](https://shopify.dev/docs/apps/auth/installation#shopify-managed-installation)
+with [token exchange](#token-exchange) instead of the authorization code grant flow.
 
-1. Start the process by calling [shopify.auth.begin](../reference/auth/begin.md) to redirect the merchant to Shopify, to ask for permission to install the app.
-1. Return the merchant to your app once they approve the app installation, by calling [shopify.auth.callback](../reference/auth/callback.md) to set up a session with an API access token.
+1. [Token Exchange](#token-exchange)
+    - Recommended for embedded apps
+    - Avoids redirect to your app to continue the OAuth flow
```

Comment:
Not sure if this would be clear to someone without a ton of context. Could we say this instead?

```suggestion
    - Doesn't require redirects, which makes it faster and prevents flickering when loading the app
```